Source/Core/DolphinQt2/NetPlay/NetPlayDialog.cpp

```diff
@@ -446,6 +450,41 @@ void NetPlayDialog::UpdateGUI()
     m_hostcode_action_button->setText(tr("Copy"));
     m_hostcode_action_button->setEnabled(true);
   }
+
+  if (m_old_player_count != player_count)
+  {
+    if (m_use_traversal)
+    {
+      const auto host_id = g_TraversalClient->GetHostID();
+      Discord::UpdateDiscordPresence(player_count, Discord::SecretType::RoomID,
+                                     std::string(host_id.begin(), host_id.end()));
+    }
+    else
+    {
+      // Temporary soluation
+      // To Do: Don't rely on a service that Dolphin devs aren't in control of. Ask one of the
+      // project managers about this.
+
+      Common::HttpRequest request;
+      Common::HttpRequest::Response response = request.Get("https://www.myexternalip.com/raw");
```

Comment:
As I mentioned before, we should use something from dolphin-emu.org for this, not an external service. Ask @delroth about that one.